File: src/main/java/com/example/onemoretimejava/basic/HcfAndLcm.java
package com.example.onemoretimejava.basic;

public class HcfAndLcm {
    public static void main(String[] args) {
        int a = 12;
        int b = 18;
        findHcf(a, b);
        findLcm(a, b);
    }

    static int findHcf(int a, int b) {
        int gcd = 1;
        int max = Math.max(a, b);
        for (int i = 1; i <= max; i++) {
            if (a % i == 0 && b % i == 0) {
                gcd = i;
            }
        }
        System.out.println("hcf: " + gcd);
        return gcd;
    }

    static int findLcm(int a, int b) {
        int lcm = 1;
        lcm = a * b / findHcf(a, b);
        System.out.println("lcm: " + lcm);
        return lcm;
    }
}

File: src/main/java/com/example/onemoretimekotlin/basic/HcfAndLcm.kt
package com.example.onemoretimekotlin.basic

fun main(args: Array<String>) {
    val num1 = 12
    val num2 = 18
    val hcf = findHcf(num1, num2)
    val lcm = findLcm(num1, num2)
    println("hcf: $hcf")
    println("lcm: $lcm")
}

fun findHcf(a: Int, b: Int): Int {
    var gcd = 1;
    val max = Math.max(a, b)
    for (i in 1..max) {
        if (a % i == 0 && b % i == 0) {
            gcd = i
        }

    }

    return gcd
}

fun findLcm(a: Int, b: Int): Int {
    var lcm = 1
    lcm = a * b / findHcf(a, b)

    return lcm
}